### ISSUE TITLE
[Question] Why am I getting 400 "Bad Request" Error when sending messages on Messenger channel?

### ISSUE THREAD

**Issue by JyothishG** [Microsoft/BotBuilder#3809]
## Bot Info 
* App ID: e402d5ef-bcd7-401d-9095-3d6808515c7f
* SDK Platform: Node.js
* Active Channels: messenger
* Deployment Environment: other 

## Issue Description

Error: Request to 'https://facebook.botframework.com/v3/conversations/15*********57-506674939684691/activities/Bf2crhG2pPo' failed: [400] Bad Request

at Request._callback (/app/node_modules/botbuilder/lib/bots/ChatConnector.js:490)
at Request.self.callback (request.js:186)
at emitTwo (events.js:126)
at Request.emit (events.js:214)
at Request.<anonymous> (/app/node_modules/request/request.js:1163)
at emitOne (events.js:116)
at Request.emit (events.js:211)
at IncomingMessage.<anonymous> (/app/node_modules/request/request.js:1085)
at Object.onceWrapper (events.js:313)
at emitNone (events.js:111)
at IncomingMessage.emit (events.js:208)
at endReadableNT (_stream_readable.js:1056)
at _combinedTickCallback (internal/process/next_tick.js:138)
at process._tickDomainCallback (next_tick.js:218)

## Code Example

This issue happens **sometimes**. The failure rate is about 15/100. I have no idea where the issue or what it is.

Thanks and regards
Jyothish G

**Comment by nwhitmont** [comment]
Hi @JyothishG - Thanks for reporting the issue. 

Do you have a `conversationId` and time that the error(s) happened? This could help with rooting out the issue in our logs.

For reference, what was the bot doing when the error happened? Were you sending text only, or messages with media attachments?  

For reference, can you post the bot code that reproduces this issue.

**Comment by nwhitmont** [comment]
any updates @JyothishG ?

**Comment by JyothishG** [comment]
Error: Request to 'https://facebook.botframework.com/v3/conversations/1526557930767857-506674939684691/activities/Bf2crhG2pPo' failed: [400] Bad Request

This was the error log that we got. There are several active users using the bot (several different messages are being sent to people who are interacting with the bot at a time). So I can't pinpoint what was the message that tried to send and fails with this error. Also I got an error related to postback buttons (see this : https://github.com/Microsoft/BotBuilder/issues/3810). 

I've got the error messages in my bot framework channel issues. They are:

{"error":{"message":"(#100) The parameter text is required","type":"OAuthException","code":100,"fbtrace_id":"BH5nxLIeIK4"}}

{"error":{"message":"(#551) This person isn't available right now.","type":"OAuthException","code":551,"error_subcode":1545041,"fbtrace_id":"E6itjK373lZ"}}

{"error":{"message":"(#-1) Unexpected internal error","type":"OAuthException","code":-1,"error_subcode":2018012,"fbtrace_id":"A4vJbUp+WiI"}}
